### PR TITLE
Expanded regex for issue scanning [semver:minor]

### PR DESCRIPTION
**Pull request by KyleTryon**
User expressed failure matching jira issues that begin with letters+numbers like so: `MG01-115`
The user also provided this suggested regex: `[A-Z0-9]{2,30}-[0-9]`

### CODE REVIEW

File: src/commands/notify.yml [hunk 1]
```diff
@@ -66,7 +66,7 @@ steps:
           -o /tmp/job_info.json \
           https://circleci.com/api/v1.1/project/${VCS_TYPE}/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/${CIRCLE_BUILD_NUM}
           # see https://jqplay.org/s/TNq7c5ctot
-          ISSUE_KEYS=$(cat /tmp/job_info.json | jq '[.all_commit_details[].subject | scan("([A-Z]{2,30}-[0-9]+)")   | .[] ] + [.all_commit_details[].branch | scan("([A-Z]{2,30}-[0-9]+)")   | .[] ]')
+          ISSUE_KEYS=$(cat /tmp/job_info.json | jq '[.all_commit_details[].subject | scan("([A-Z0-9]{2,30}-[0-9]+)")   | .[] ] + [.all_commit_details[].branch | scan("([A-Z]{2,30}-[0-9]+)")   | .[] ]')
```
Comment (nullcode) [inline src/commands/notify.yml]: Hey, I'm the one who reported this problem. You forgot to adjust the second regex with the same pattern here for the branch name.

**Comment by eddiewebb** [comment]
@KyleTryon / @nullcode - the original regex was based on Jira's default project key rules.

It seems admins can customize to allow #s or other changes. - https://confluence.atlassian.com/adminjiraserver073/changing-the-project-key-format-861253229.html

Rather than hardcoding it here, we should move the regexp to a parameter and set a reasonable default.

**Comment by nullcode** [comment]
@eddiewebb right, we are actually using Jira Cloud and the regex is not even given there, they just describe it like this in the docs: [Your project's key must be at least 2 characters long and contain only uppercase letters.](https://confluence.atlassian.com/jiracorecloud/editing-a-project-s-details-938021482.html) 
The parameter option would be a nice solution.

**Comment by KyleTryon** [comment]
Will submit an update to include parameter with the original default value.

File: src/commands/notify.yml [hunk 2]
```diff
@@ -66,7 +70,7 @@ steps:
           -o /tmp/job_info.json \
           https://circleci.com/api/v1.1/project/${VCS_TYPE}/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/${CIRCLE_BUILD_NUM}
           # see https://jqplay.org/s/TNq7c5ctot
-          ISSUE_KEYS=$(cat /tmp/job_info.json | jq '[.all_commit_details[].subject | scan("([A-Z0-9]{2,30}-[0-9]+)")   | .[] ] + [.all_commit_details[].branch | scan("([A-Z]{2,30}-[0-9]+)")   | .[] ]')
+          ISSUE_KEYS=$(cat /tmp/job_info.json | jq '[.all_commit_details[].subject | scan("(<<parameters.issue_regexp>>)")   | .[] ] + [.all_commit_details[].branch | scan("([A-Z]{2,30}-[0-9]+)")   | .[] ]')
```
Comment (eddiewebb) [inline src/commands/notify.yml]: Need to also replace the 2nd occurence on this line (1st looks at subject, 2nd look sat branch)

**Comment by eddiewebb** [comment]
Testing against dev version looks good, merging this.

**Comment by eddiewebb** [comment]
This is available as `circleci/jira@1.1.0`

**Pull request merged**